CoEpi/ble/ContactReceivedHandler.swift

```diff
@@ -13,11 +14,20 @@ class ContactReceivedHandler {
 
     // TODO review CEN generation timing
     func provideMyCen() -> Data {
-        let currentCENKey = cenKeyRepo.generateAndStoreCENKey()
-        return cenLogic.generateCen(CENKey: currentCENKey.cenKey)
-        //*** Scenario 1: https://docs.google.com/document/d/1f65V3PI214-uYfZLUZtm55kdVwoazIMqGJrxcYNI4eg/edit#
-        // iOS - Central + iOS - Peripheral -- so commenting out addNewContact
-        //addNewContactEvent(with: identifier)
+        switch cenKeyRepo.generateAndStoreCENKey() {
+        case .success(let key):
+            return cenLogic.generateCen(CENKey: key.cenKey)
+            //*** Scenario 1: https://docs.google.com/document/d/1f65V3PI214-uYfZLUZtm55kdVwoazIMqGJrxcYNI4eg/edit#
+            // iOS - Central + iOS - Peripheral -- so commenting out addNewContact
+            //addNewContactEvent(with: identifier)
+
+        case .failure(let error):
+            os_log("Couldn't generate CEN key: %@", type: .error, "\(error)")
+            // TODO clarify with BT lib how to handle error.
```

Comment:
App running without CEN key does not make much sense.

Reply:
Yep. We will address the edge cases / TODOs once the core functionality is finished.